### ISSUE TITLE
Typescript 4.8.2 and loadFromPivotTable

### ISSUE THREAD

**Issue by roderik** [mikro-orm/mikro-orm#3417]
**Describe the bug**

Reported a few times before its official release, but now it has been released and dependabot opend a PR on our repo

- https://github.com/mikro-orm/mikro-orm/issues/3367
- https://github.com/mikro-orm/mikro-orm/issues/3367

**Stack trace**
```
error TS2344: Type 'MongoDriver' does not satisfy the constraint 'IDatabaseDriver<Connection>'.
             Types of property 'loadFromPivotTable' are incompatible.
               Type '<T, O>(prop: EntityProperty<any>, owners: Primary<O>[][], where?: FilterQuery<T> | undefined, orderBy?: QueryOrderMap<T>[] | undefined, ctx?: any, options?: FindOptions<...> | undefined) => Promise<...>' is not assignable to type '<T extends AnyEntity<T>, O extends AnyEntity<O>>(prop: EntityProperty<any>, owners: Primary<O>[][], where?: FilterQuery<T> | undefined, orderBy?: QueryOrderMap<...>[] | undefined, ctx?: any, options?: FindOptions<...> | undefined) => Promise<...>'.
                 Types of parameters 'options' and 'options' are incompatible.
                   Type 'FindOptions<T, never> | undefined' is not assignable to type 'FindOptions<T | ((Primary<T> extends string ? string | RegExp : Primary<T> extends Date ? string | Date : Primary<T>) & {}), never> | undefined'.
                     Type 'FindOptions<T, never>' is not assignable to type 'FindOptions<T | ((Primary<T> extends string ? string | RegExp : Primary<T> extends Date ? string | Date : Primary<T>) & {}), never>'.
                       Types of property 'fields' are incompatible.
                         Type 'readonly EntityField<T, never>[] | undefined' is not assignable to type 'readonly EntityField<T | ((Primary<T> extends string ? string | RegExp : Primary<T> extends Date ? string | Date : Primary<T>) & {}), never>[] | undefined'.
                           Type 'readonly EntityField<T, never>[]' is not assignable to type 'readonly EntityField<T | ((Primary<T> extends string ? string | RegExp : Primary<T> extends Date ? string | Date : Primary<T>) & {}), never>[]'.
                             Type 'EntityField<T, never>' is not assignable to type 'EntityField<T | ((Primary<T> extends string ? string | RegExp : Primary<T> extends Date ? string | Date : Primary<T>) & {}), never>'.
                               Type 'keyof T' is not assignable to type 'EntityField<T | ((Primary<T> extends string ? string | RegExp : Primary<T> extends Date ? string | Date : Primary<T>) & {}), never>'.
                                 Type 'string | number | symbol' is not assignable to type 'EntityField<T | ((Primary<T> extends string ? string | RegExp : Primary<T> extends Date ? string | Date : Primary<T>) & {}), never>'.
                                   Type 'string' is not assignable to type 'EntityField<T | ((Primary<T> extends string ? string | RegExp : Primary<T> extends Date ? string | Date : Primary<T>) & {}), never>'.
```

**To Reproduce**
1. Upgrade typescript

**Comment by B4nan** [comment]
Yes, its out just few hours :] The release is _very_ breaking for our codebase, you can see how many changes were required to support it partially in #3389. This might take some time to finish.

**Comment by roderik** [comment]
auch!

**Comment by B4nan** [comment]
One of the issues TS reports is also coming from knex, cc @kibertoad 

```
$ tsc --noEmit --project tests/tsconfig.json
node_modules/knex/types/index.d.ts:453:33 - error TS2344: Type 'TRecord | TResult' does not satisfy the constraint '{}'.
  Type 'TResult' is not assignable to type '{}'.

453             | Knex.QueryBuilder<TRecord | TResult>
                                    ~~~~~~~~~~~~~~~~~
```

**Comment by kibertoad** [comment]
It was fixed in https://github.com/knex/knex/pull/5279, I believe, but not yet released. I'll publish a new version soon

**Comment by B4nan** [comment]
So once knex is released, #3389 should be good to go. I am sure this will break some projects on type level, but what can we do :]

**Comment by B4nan** [comment]
@kibertoad could we have the fix published please?

**Comment by kibertoad** [comment]
yes, will do it tonight

**Comment by kibertoad** [comment]
@B4nan released in 2.3.0, please let me know if it works as expected

**Comment by B4nan** [comment]
Yes, perfect, thanks!
